ntbs-service/Services/NotificationChangesService.cs

```diff
@@ -288,8 +300,30 @@ private static string MapAction(AuditLog log)
             }
         }
 
+        private static bool LogIsServiceShare(AuditLog log)
+        {
+            return LogIsForHospitalDetailsAndHasDataWhichMatchesString(log, "\"NewValue\":\"TBS\\d{4}\"");
```

Comment:
It's very dangerous here to make this assumption about the format of a TB service code.

I think it would be better to try to deserialize the JSON here and then just check whether the `NewValue` or `OriginalValue` properties are set.

Reply:
I have moved this to now happen in GetCanonicalLogs and not in these mapping functions (as the logs were all being checked twice for the same conditions). It now deserializes the update, and then checks for the relevant values/keys